bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/internal/NetworkUtils.java

```diff
@@ -0,0 +1,143 @@
+     *
+     * @param host hostname or ip address of the host to check for availability
+     * @param timeout time in milliseconds for the check to timeout
+     * @param logger the logger to use for logging any issues
+     * @throws IOException if there is an unexpected error
+     * @throws BroadlinkHostNotReachableException when the device is not responding
+     */
+
+    public static void hostAvailabilityCheck(@Nullable String host, int timeout, Logger logger)
+            throws IOException, BroadlinkHostNotReachableException {
+        InetAddress address = InetAddress.getByName(host);
+        if (!address.isReachable(timeout)) {
+            throw new BroadlinkHostNotReachableException("Cannot reach " + host);
+        }
+    }
+
+    /**
+     * Finds an InetAddress that is associated to a non-loopback device.
+     *
+     * @return null, if there is no non-loopback device address, otherwise the InetAddress associated to a non-loopback
+     *         device
+     * @throws SocketException thrown when no socket can be opened.
+     */
+    public static @Nullable InetAddress findNonLoopbackAddress() throws SocketException {
+        for (InetAddress address : NetUtil.getAllInterfaceAddresses().stream()
+                .filter(a -> a.getAddress() instanceof Inet4Address).map(a -> a.getAddress()).toList()) {
+            if (address.isSiteLocalAddress()) {
+                return address;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Find the address of the local lan host
+     *
+     * @return InetAddress of the local lan host
+     * @throws UnknownHostException if no local lan address can be found
+     */
+    public static InetAddress getLocalHostLANAddress() throws UnknownHostException {
+        try {
+            InetAddress candidateAddress = findNonLoopbackAddress();
+
+            if (candidateAddress != null) {
+                return candidateAddress;
+            }
+            InetAddress jdkSuppliedAddress = InetAddress.getLocalHost();
+            if (jdkSuppliedAddress == null) {
+                throw new UnknownHostException("The JDK InetAddress.getLocalHost() method unexpectedly returned null.");
+            } else {
+                return jdkSuppliedAddress;
+            }
+        } catch (Exception e) {
+            UnknownHostException unknownHostException = new UnknownHostException(
+                    (new StringBuilder("Failed to determine LAN address: ")).append(e).toString());
+            unknownHostException.initCause(e);
+            throw unknownHostException;
+        }
+    }
+
+    /**
+     * Randomly find a free port on the host in the defined range
+     *
+     * @param host The address of the host to find a free port on
+     * @param from port number of the start of the range
+     * @param to port number of the end of the range
+     * @return number of the available port
+     */
+    public static int nextFreePort(InetAddress host, int from, int to) {
```

Comment:
You might want to add check `from < to`

Another issue is what if there is no free port, this method will end up in and endless loop. Don't you want to add a timeout liek after 30 seconds it will quit the loop no matter if it was finished or not?

Reply:
Added a check. I think you have significant bigger issues when there is no port free anymore on your device :-). Anyway, I have added a timeout.